File: app/ui/PlaySession.java
package ui;

import models.IEndUser;
import models.IOrder;
import models.ISession;
import models.dummy.DummyOrder;
import models.dummy.EndUser;
import play.mvc.Http;

import javax.persistence.criteria.Order;
import java.util.Dictionary;
import java.util.HashMap;

/**
 * Created by Beemen on 30/07/2015.
 */
public class PlaySession implements ISession {

    public IOrder CurrentOrder(){
        String key = orderKey();
        Object ret = play.cache.Cache.get(key);
        if(ret == null){
            NewOrder();
            ret = play.cache.Cache.get(key);
        }
        return (IOrder)ret;
    }

    @Override
    public void NewOrder() {
        play.cache.Cache.set(
                orderKey(),
                new DummyOrder(){{
                    User(this.User());
                }}
        );
    }

    public String SessionId(){
        Http.Session session = Http.Context.current().session();

        String uuid = session.get("uuid");
        if (uuid == null) {
            uuid = java.util.UUID.randomUUID().toString();
            session.put("uuid", uuid);
        }
        return uuid;
    }

    String orderKey(){
        return SessionId() + "_CurrentOrder";
    }

    public final HashMap<String,IEndUser> _Users = new HashMap<>();
    public IEndUser User(){
        _Users.computeIfAbsent(
                SessionId(),
                k ->  new EndUser(){{
                    Name(SessionId());
                    UniqueId(SessionId());
                }});

        return  _Users.get(SessionId());
    }

    private play.mvc.Http.Session _Session;

    public static PlaySession Current(){
        PlaySession ret = new PlaySession();
        ret._Session = play.mvc.Http.Context.current().session();
        return ret;
    }


}

File: UI/PlaySession.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using end_user_gui.Models;
using end_user_gui.Models.dummy;

namespace end_user_gui.UI
{
    public class PlaySession : ISession
    {
        private static Dictionary<string, object> _Cache = new Dictionary<string, object>();

        public IOrder CurrentOrder
        {
            get
            {
                String key = OrderKey;
                if (!_Cache.ContainsKey(key))
                {
                    NewOrder();
                }
                Object ret = _Cache[key];
                return (IOrder)ret;
            }
        }

        public void NewOrder()
        {
            _Cache[OrderKey] = new DummyOrder() { User = this.User };
        }

        public String SessionId
        {
            get
            {
                var session = System.Web.HttpContext.Current.Application;

                object uuid = session["uuid"];
                if (uuid == null)
                {
                    uuid = Guid.NewGuid().ToString();
                    session["uuid"] = uuid;
                }
                return uuid as string;
            }
        }

        String OrderKey
        {
            get
            {
                return SessionId + "_CurrentOrder";
            }
        }

        public static readonly Dictionary<String, EndUser> _Users = new Dictionary<string, EndUser>();
        public EndUser User
        {
            get
            {
                if (!_Users.ContainsKey(SessionId))
                {
                    _Users[SessionId] = new EndUser() { Name = SessionId, UniqueId = SessionId };
                }

                return _Users[SessionId];
            }
        }

        private System.Web.SessionState.HttpSessionState _Session;

        public static PlaySession Current
        {
            get
            {
                PlaySession ret = new PlaySession();
                ret._Session = System.Web.HttpContext.Current.Session;
                return ret;
            }
        }
    }
}